builtInFunctions/esdtModifyRoyalties.go

```diff
@@ -0,0 +1,125 @@
+package builtInFunctions
+
+import (
+	"bytes"
+	"math/big"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+)
+
+const (
+	tokenIDIndex      = 0
+	nonceIndex        = 1
+	newRoyaltiesIndex = 2
+)
+
+type esdtModifyRoyalties struct {
+	baseActiveHandler
+	globalSettingsHandler vmcommon.GlobalMetadataHandler
+	storageHandler        vmcommon.ESDTNFTStorageHandler
+	rolesHandler          vmcommon.ESDTRoleHandler
+	accounts              vmcommon.AccountsAdapter
+}
+
+// NewESDTModifyRoyaltiesFunc returns the esdt modify royalties built-in function component
+func NewESDTModifyRoyaltiesFunc(
+	accounts vmcommon.AccountsAdapter,
+	globalSettingsHandler vmcommon.GlobalMetadataHandler,
+	storageHandler vmcommon.ESDTNFTStorageHandler,
+	rolesHandler vmcommon.ESDTRoleHandler,
+	enableEpochsHandler vmcommon.EnableEpochsHandler,
+) (*esdtModifyRoyalties, error) {
+	if check.IfNil(accounts) {
+		return nil, ErrNilAccountsAdapter
+	}
+	if check.IfNil(globalSettingsHandler) {
+		return nil, ErrNilGlobalSettingsHandler
+	}
+	if check.IfNil(enableEpochsHandler) {
+		return nil, ErrNilEnableEpochsHandler
+	}
+	if check.IfNil(storageHandler) {
+		return nil, ErrNilESDTNFTStorageHandler
+	}
+	if check.IfNil(rolesHandler) {
+		return nil, ErrNilRolesHandler
+	}
+
+	e := &esdtModifyRoyalties{
+		accounts:              accounts,
+		globalSettingsHandler: globalSettingsHandler,
+		storageHandler:        storageHandler,
+		rolesHandler:          rolesHandler,
+	}
+
+	e.baseActiveHandler.activeHandler = enableEpochsHandler.IsDynamicESDTEnabled
+
+	return e, nil
+}
+
+// ProcessBuiltinFunction saves the token type in the system account
+func (e *esdtModifyRoyalties) ProcessBuiltinFunction(acntSnd, _ vmcommon.UserAccountHandler, vmInput *vmcommon.ContractCallInput) (*vmcommon.VMOutput, error) {
+	if vmInput == nil {
+		return nil, ErrNilVmInput
+	}
+	if vmInput.CallValue == nil {
+		return nil, ErrNilValue
+	}
+	if vmInput.CallValue.Cmp(zero) != 0 {
+		return nil, ErrBuiltInFunctionCalledWithValue
+	}
+	if len(vmInput.Arguments) != 3 {
+		return nil, ErrInvalidNumberOfArguments
+	}
+	if !bytes.Equal(vmInput.CallerAddr, vmInput.RecipientAddr) {
+		return nil, ErrInvalidRcvAddr
+	}
+	if check.IfNil(acntSnd) {
+		return nil, ErrNilUserAccount
+	}
+	if !e.baseActiveHandler.IsActive() {
+		return nil, ErrBuiltInFunctionIsNotActive
+	}
+	// TODO check and consume gas
```

Comment:
funcGasCost everywhere, like in other built in functions.

Reply:
Done.